### DISCUSSION TITLE
Async to wait for element on screen

### DISCUSSION THREAD

**Discussion by mabitten** [gilbarbara/react-joyride#1034] [Q&A] [answered]
How do you solve the problem today when the element will still be rendered on the screen?

I'm using react-joyride and in one of the steps, the helmet whose id is set takes a few seconds to appear on the screen when it reaches the step, instead, it skips to the next one. How do you do today? Is there any option to "wait/watch" for the id element to appear on the screen and then proceed to the next step?

**Comment by gilbarbara** [comment, marked as answer]
Hey @mabitten 

You'll need to use the controlled mode (setting a stepIndex that you must control yourself).
Check the source code for the [Controlled](https://codesandbox.io/p/devbox/github/gilbarbara/react-joyride-demo/tree/main/?file=%2Fsrc%2FControlled%2Findex.tsx%3A154%2C1-155%2C1) demo

**Comment by mabitten** [comment]
Hi @gilbarbara 

Perfect, I saw it before. But I think we write a lot of code in my case.
But It works.

Would it be interesting to create something asynchronous/observable to meet needs where I just need to wait for the element to be on the screen? I'm thinking about forking and implementing

**Comment by gilbarbara** [comment]
Hey @mabitten 
Async support is planned for v3, but I don't want to add observables. Most people won't need them, and it just increases the bundle size. The callback and stepIndex were designed for these cases.
